### ISSUE TITLE
Add padding in batched graphs 

### ISSUE THREAD

**Issue by francescofarina** [mlx-graphs/mlx-graphs#93]
Let's consider whether it makes sense to add padding in batched graphs so that each batch of the dataloader has the same size. In this way we can better exploit compilation.

More details in #92.

**Comment by francescofarina** [comment]
Consider the following two edge indices
```
edge_index_1 = mx.array([[0],[1]])
edge_index_2 = mx.array([[0, 1], [1, 2]])
```
Collating them currently produce a new edge index
```
collated_edge_index = mx.array([[0, 2, 3],[1, 3, 4]])
```
Adding a padding to the first one would result in
```
collated_edge_index = mx.array([[0, X, 2, 3],[1, X, 3, 4]])
```
The question is what value to assign to `X`? 
We can't use `0` as that's a node index and we allow self-loops. 
We could potentially use `-1` but then we'd need to prune those edges when the edge index needs to be used.

In general it doesn't seem obvious to find a solution that doesn't add any overhead when the collated graph needs to be used.

**Comment by TristanBilot** [comment]
`-1` would be a good solution as we can add an extra "dummy" node with an all-zero (no-op) node feature vector. If we want to retrieve the original graphs, we'll just have to remove `-1` nodes (e.g. with mx.where). In a first time, we can test if this effectively improves performance, and we also need to evaluate beforehand if the batching is feasible (e.g. by checking the size of the largest graph).

Also, do you think it's better to pad per graph or to pad per batch?
For example, we can batch graphs together until a given number of edges threshold is reached and complete with no-op edges at the end of the batch. I think this solution may be more robust to different sizes of graphs but this would require to set a number of edges instead of number of graphs. We can also provide both options.

**Comment by francescofarina** [comment]
Actually I think the solution you propose of pdding the batch may be the best one - with that we'd only add padding at the end and keep track of the last element of the actual collated graphs, hence maintaining most of the logic we already have. Wdyt?

**Comment by TristanBilot** [comment]
Exactly, it would be much easier. We can start first trials with this logic.
